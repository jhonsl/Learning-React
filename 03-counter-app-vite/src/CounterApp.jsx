import PropTypes from 'prop-types'
import { useState } from 'react'

export const CounterApp = ({ value }) => {

  const [ counter, setCounter ] = useState( value );

  const incrementCounter = () => setCounter( (value) => value + 1 );
  // const incrementCounter = () => setCounter( counter + 1 );
  const decrementCounter = () => setCounter( (value) => value - 1 );
  // const decrementCounter = () => setCounter( counter - 1 );
  const resetCounter = () => setCounter( value );

  return (
    <>
      <h1>Counter App</h1>
      <h2 data-testid="title">El valor es: { counter }</h2>
      <button aria-label='btn-add' onClick={ incrementCounter }>
        +1
      </button>
      <button aria-label='btn-rest' onClick={ decrementCounter }>
        -1
      </button>
      <button aria-label='btn-reset' onClick={ resetCounter }>
        Reset
      </button>
    </>
  )
}

CounterApp.propTypes = {
  value: PropTypes.number.isRequired
}