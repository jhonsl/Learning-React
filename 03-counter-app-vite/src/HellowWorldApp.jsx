import PropTypes from 'prop-types';

const message = {
    title: 'hola mundo',
    subTitle: 'mi nombre es jhon'
};
const numeros = [1,2,3,4,5,6,7,8,9];
const sumar = (numero1, numero2) => numero1 + numero2;

export const HellowWorldApp = ({ title, subTitle }) => {
    return (
        <>
            <h1>{ message.title }</h1>
            <code>{ JSON.stringify(message) }</code>
            <p>{ message.subTitle }</p>
            <p>{ numeros }</p>

            <hr />

            <h2>El resultado de la suma es: { sumar(4,5) }</h2>
            <h2>{ title }</h2>
            <h3>{ subTitle }</h3>
        </>
    )
}

HellowWorldApp.defaultProps = {
    title: 'mil a√±os de soledad',
    subTitle: 2002
}

HellowWorldApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.number.isRequired
}