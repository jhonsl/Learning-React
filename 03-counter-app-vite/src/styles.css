html, body {
    background-color: #21232A;
    color: white;
    font-family: Helvetica, Arial, sans-serif;
    font-size: 1.3rem;
    padding: 70px;
}

button {
    padding: 10px 15px 10px 15px;
    margin-right: 20px;
    font-size: 1.3rem;
}