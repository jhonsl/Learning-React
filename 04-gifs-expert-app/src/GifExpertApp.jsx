import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {

  const [categories, setCategories] = useState(['Dragon Ball']);
  const onAddCategory = ( category ) => {

    if( !categories.includes( category ) ) {
      setCategories([...categories, category])
    }
  };
 
  return (
    <>
      <h1>GifsExpertApp</h1> 

      <AddCategory onNewCategory={ onAddCategory } />

      { 
        categories.map( category => (
          <GifGrid 
            key={ category } 
            category={ category }
          />
        )) 
      }
    </>
  )
}

export default GifExpertApp;