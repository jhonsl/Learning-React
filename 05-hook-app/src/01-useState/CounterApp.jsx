import { useState } from "react"

export const CounterApp = () => {

    const [{ counter1, ...rest }, setCounter] = useState({
        counter1: 10,
        counter2: 20,
        counter3: 30
    })

    return (
        <>
            <h1>Counter1: { counter1 }</h1>
            <h1>Counter2: { rest.counter2 }</h1>
            <h1>Counter3: { rest.counter3 }</h1>

            <hr />

            <button 
                className="btn"
                onClick={() => setCounter({
                    ...rest,
                    counter1: counter1 + 1
                })}
            >
                +1
            </button>
        </>
    )
}
