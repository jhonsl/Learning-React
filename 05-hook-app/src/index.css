body {
  padding: 20px;
}

button {
  margin-right: 10px;
}

/* a {
  border-right: 2px solid black;
  margin-left: 1%;
  padding-right: 1%;
} */