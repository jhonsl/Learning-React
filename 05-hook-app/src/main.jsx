import React from 'react'
import ReactDOM from 'react-dom/client'
import { Padre } from './07-tarea-memo/Padre'

import './index.css'

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <Padre />
  </React.StrictMode>
)
