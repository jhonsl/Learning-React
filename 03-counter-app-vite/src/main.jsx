import React from 'react';
import ReactDOM from 'react-dom/client'
import { CounterApp } from './CounterApp'
import { FirstApp } from './FirstApp';
import './styles.css'

ReactDOM.createRoot( document.getElementById('root') ).render(
    <React.StrictMode>
        <CounterApp value={10}/>
        {/* <FirstApp title='Esto es un titulo'/> */}
    </React.StrictMode>
)  