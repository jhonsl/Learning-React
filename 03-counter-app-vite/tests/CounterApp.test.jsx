import { fireEvent, render, screen } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";

describe('Pruebas en <CounterApp />', () => {

    const initialValue = 100;
    test('debe de hacer match con el Snapshot', () => {
        const { container } = render( <CounterApp value={initialValue}/> );
        expect( container ).toMatchSnapshot();
    });

    test('debe de mostar el valor inicial de 100', () => {
        render( <CounterApp value={initialValue}/> );
        expect( screen.getByTestId('title').innerHTML ).toContain( String(initialValue) );
        expect( screen.getByRole('heading', { level: 2 }).innerHTML ).toContain( String(initialValue) );
    });

    test('debe de incrementar el valor inicial con el boton +1', () => {
        render( <CounterApp value={initialValue}/> );
        fireEvent.click( screen.getByRole('button', { name: 'btn-add' }) );
        expect( screen.getByText(`El valor es: ${initialValue + 1}`) ).toBeTruthy();
    });

    test('debe de decrementar el valor inicial con el boton -1', () => {
        render( <CounterApp value={initialValue}/> );
        fireEvent.click( screen.getByRole('button', { name: 'btn-rest' }) );
        expect( screen.getByText(`El valor es: ${initialValue - 1}`) ).toBeTruthy();
    });

    test('debe de funcionar el boton de reset', () => {
        render( <CounterApp value={initialValue}/> );
        for(let i = 0; i < 3; i++) {
            fireEvent.click( screen.getByRole('button', { name: 'btn-add' }) );
        }
        expect( screen.getByText(`El valor es: ${initialValue + 3}`) ).toBeTruthy();
        fireEvent.click( screen.getByRole('button', { name: 'btn-reset' }) );
        expect( screen.getByText(`El valor es: ${initialValue}`) ).toBeTruthy();
    });
});