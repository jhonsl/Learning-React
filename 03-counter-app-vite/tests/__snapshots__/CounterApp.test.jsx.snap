// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp /> debe de hacer match con el Snapshot 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h2
    data-testid="title"
  >
    El valor es: 
    100
  </h2>
  <button
    aria-label="btn-add"
  >
    +1
  </button>
  <button
    aria-label="btn-rest"
  >
    -1
  </button>
  <button
    aria-label="btn-reset"
  >
    Reset
  </button>
</div>
`;
